src/test/java/net/snowflake/client/jdbc/ConnectionLatestIT.java

```diff
@@ -966,7 +966,13 @@ public void testAuthenticatorEndpointWithDashInAccountName() throws Exception {
     postRequest.addHeader("accept", "application/json");
 
     String theString =
-        HttpUtil.executeGeneralRequest(postRequest, 60, 0, 0, 0, new HttpClientSettingsKey(null));
+        HttpUtil.executeGeneralRequest(
```

Comment:
Any way to have test cases actually inspect request headers to check for the user-agent construction?

Reply:
The user agent headers are added as part of HttpClient's Execution chain and are applied very late during the request execution. That part of the code is not easily testable.